### PR TITLE
feat: change global logger to enforcer

### PR DESCRIPTION
**Pull request by kocoler**
- Fix print stdout log without control.
- Change global logger to enforcer. Every enforcer as well as its role manager and model could use its own logger.

- Fix: https://github.com/casbin/casbin/issues/629

### CODE REVIEW

**Comment by kocoler** [comment]
Fix #629

**Comment by hsluoyz** [comment]
@kocoler please fix DCO.

**Comment by hsluoyz** [comment]
@kocoler plz make it draft if not ready

**Comment by kocoler** [comment]
@hsluoyz plz review.

File: log/default_logger.go [hunk 1]
```diff
@@ -14,39 +14,44 @@
...
-import (
-	"log"
-	"sync/atomic"
-)
+import "log"
 
 // DefaultLogger is the implementation for a Logger using golang log.
 type DefaultLogger struct {
-	enable int32
+	enable bool
```
Comment (hsluoyz) [inline log/default_logger.go:21]: enabled

File: enforcer.go [hunk 2]
```diff
@@ -329,7 +345,14 @@ func (e *Enforcer) EnableEnforce(enable bool) {
 
 // EnableLog changes whether Casbin will log messages to the Logger.
 func (e *Enforcer) EnableLog(enable bool) {
-	log.GetLogger().EnableLog(enable)
+	//log.GetLogger().EnableLog(enable)
```
Comment (hsluoyz) [inline enforcer.go:348]: Remove all comments.

File: enforcer.go [hunk 3]
```diff
@@ -173,6 +185,8 @@ func (e *Enforcer) initialize() {
 	e.autoSave = true
 	e.autoBuildRoleLinks = true
 	e.autoNotifyWatcher = true
+
+	e.rm.SetRMLogger(e.logger)
```
Comment (hsluoyz) [inline enforcer.go:189]: Change ALL to SetLogger, GetLogger

File: model/model.go [hunk 4]
```diff
@@ -94,9 +95,28 @@ func loadSection(model Model, cfg config.ConfigInterface, sec string) {
 	}
 }
 
+// SetModelLogger sets the model's logger.
+func (model Model) SetModelLogger(logger log.Logger) {
+	as := AssertionMap{}
+	as["logger"] = &Assertion{logger: &logger}
+	model["logger"] = as
+	for _, v := range model {
```
Comment (hsluoyz) [inline model/model.go:103]: This code is so ugly, please change.

**Review by hsluoyz — CHANGES_REQUESTED** [review]
Please fix.

File: enforcer.go [hunk 5]
```diff
@@ -329,7 +345,12 @@ func (e *Enforcer) EnableEnforce(enable bool) {
 
 // EnableLog changes whether Casbin will log messages to the Logger.
 func (e *Enforcer) EnableLog(enable bool) {
-	log.GetLogger().EnableLog(enable)
+	e.logger.EnableLog(enable)
+}
+
+// GetLoggerEnableStatus returns the current logger status.
+func (e *Enforcer) GetLoggerEnableStatus() bool {
```
Comment (hsluoyz) [inline enforcer.go:352]: GetLoggerEnabled

**Review by hsluoyz — APPROVED** [review]
lgtm

**Pull request merged**